File: sdk/python/src/chainlogistics_sdk/services/stats.py
"""Stats service for accessing statistics and analytics."""

from ..client import ChainLogisticsClient
from ..models import DbHealthResponse, GlobalStats, HealthResponse


class StatsService:
    """Service for accessing statistics and analytics."""
    
    def __init__(self, client: ChainLogisticsClient):
        """Initialize the service.
        
        Args:
            client: ChainLogistics client instance
        """
        self.client = client
    
    def get_global(self) -> GlobalStats:
        """Get global statistics.
        
        Returns:
            Global statistics
        """
        data = self.client.get("api/v1/stats")
        return GlobalStats(**data)
    
    def health(self) -> HealthResponse:
        """Get system health status.
        
        Returns:
            Health response
        """
        data = self.client.get("health")
        return HealthResponse(**data)
    
    def db_health(self) -> DbHealthResponse:
        """Get database health status.
        
        Returns:
            Database health response
        """
        data = self.client.get("health/db")
        return DbHealthResponse(**data)

File: sdk/rust/src/stats.rs
use crate::{
    client::HttpClient,
    models::GlobalStats,
    Config, Result,
};

/// Service for accessing statistics and analytics
#[derive(Debug, Clone)]
pub struct StatsService {
    client: HttpClient,
}

impl StatsService {
    pub(crate) fn new(client: reqwest::Client, config: Config) -> Self {
        Self {
            client: HttpClient::new(client, config),
        }
    }

    /// Get global statistics
    pub async fn get_global(&self) -> Result<GlobalStats> {
        let request = self.client.get("api/v1/stats");
        self.client.execute(request).await
    }

    /// Get system health status
    pub async fn health(&self) -> Result<crate::models::HealthResponse> {
        let request = self.client.get("health");
        self.client.execute(request).await
    }

    /// Get database health status
    pub async fn db_health(&self) -> Result<crate::models::DbHealthResponse> {
        let request = self.client.get("health/db");
        self.client.execute(request).await
    }
}
